### ISSUE TITLE
 Unable to Create Dailog catche ? 

### ISSUE THREAD

**Issue by Dhiru-Dev** [QuickBlox/q-municate-services-ios#238]
When i open chat list i get only one dialog from Cache ,,,but there is lots of Chat Dialog for this user .

logs

```
` MyApp[8256:227450] Init - QMAuthService service...
MyApp[8256:227450] Init - QMChatService service...
MyApp[8256:227450] Init - QMUsersService service...
MyApp[8256:227714] [QMChatCache] Dialogs to insert 0, update 1
MyApp[8256:227714] [QMChatCache] Messages to insert 0, update 2`
```

![screen shot 2016-07-12 at 6 24 35 pm](https://cloud.githubusercontent.com/assets/16178823/16767969/9f331484-485f-11e6-9a17-0fb913a05c8f.png)

**Comment by Dhiru-Dev** [comment]
can anyone tell me how do i fetch the All dailogs and Create it's cache ..??

**Comment by Dhiru-Dev** [comment]
When i am Trying  to add in Memory my app crashes 

`[[ServicesManager instance].dialogsInMemory addChatDialogs:dialogObjects andJoin:YES];`

 Here is the Crash Logs

```
 *** Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: '-[ServicesManager dialogMemoryStorage]: unrecognized selector sent to instance 0x7f9212407a90'
*** First throw call stack:
(
    0   CoreFoundation                      0x0000000114d56e65 __exceptionPreprocess + 165
    1   libobjc.A.dylib                     0x00000001144a3deb objc_exception_throw + 48
    2   CoreFoundation                      0x0000000114d5f48d -[NSObject(NSObject) doesNotRecognizeSelector:] + 205
    3   CoreFoundation                      0x0000000114cac90a ___forwarding___ + 970
    4   CoreFoundation                      0x0000000114cac4b8 _CF_forwarding_prep_0 + 120
    5   MyApp                              0x000000010fb02b8f __27-[ChatListVC fetchChatList]_block_invoke + 303
    6   MyApp                              0x000000010fc43e0c -[TWMessageBarViewController statusBarHidden] + 277324
    7   MyApp                              0x000000010fc12b82 -[TWMessageBarViewController statusBarHidden] + 75970
    8   MyApp                              0x000000010fc13215 -[TWMessageBarViewController statusBarHidden] + 77653
    9   MyApp                              0x000000010fc70fc8 QB_AFQueryStringPairsFromKeyAndValue + 47200
    10  libdispatch.dylib                   0x00000001158e5e5d _dispatch_call_block_and_release + 12
    11  libdispatch.dylib                   0x000000011590649b _dispatch_client_callout + 8
    12  libdispatch.dylib                   0x00000001158ee2af _dispatch_main_queue_callback_4CF + 1738
    13  CoreFoundation                      0x0000000114cb6d09 __CFRUNLOOP_IS_SERVICING_THE_MAIN_DISPATCH_QUEUE__ + 9
    14  CoreFoundation                      0x0000000114c782c9 __CFRunLoopRun + 2073
    15  CoreFoundation                      0x0000000114c77828 CFRunLoopRunSpecific + 488
    16  GraphicsServices                    0x00000001168c3ad2 GSEventRunModal + 161
    17  UIKit                               0x0000000112108610 UIApplicationMain + 171
    18  MyApp                              0x000000010fa943df main + 111
    19  libdyld.dylib                       0x000000011593b92d start + 1
    20  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by Dhiru-Dev** [comment]
When  i am trying to send Dialogs 

```
*** Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'Dialog have to be in memory cache!'
*** First throw call stack:
(
    0   CoreFoundation                      0x000000011177ce65 __exceptionPreprocess + 165
    1   libobjc.A.dylib                     0x0000000110ec9deb objc_exception_throw + 48
    2   CoreFoundation                      0x000000011177ccca +[NSException raise:format:arguments:] + 106
    3   Foundation                          0x0000000110b154de -[NSAssertionHandler handleFailureInMethod:object:file:lineNumber:description:] + 198
    4   Cyrano                              0x000000010c5e9aaf -[QMChatService sendMessage:toDialogID:saveToHistory:saveToStorage:completion:] + 911
    5   Cyrano                              0x000000010c552d9b -[ChatVC didPressSendButton:withMessageText:senderId:senderDisplayName:date:] + 1147
    6   Cyrano                              0x000000010c4851e4 -[QMChatViewController messagesInputToolbar:didPressRightBarButton:] + 276
    7   Cyrano                              0x000000010c46c26d -[QMInputToolbar rightBarButtonPressed:] + 93
    8   UIKit                               0x000000010eb30194 -[UIApplication sendAction:to:from:forEvent:] + 92
    9   UIKit                               0x000000010ec9f6fc -[UIControl sendAction:to:forEvent:] + 67
    10  UIKit                               0x000000010ec9f9c8 -[UIControl _sendActionsForEvents:withEvent:] + 311
    11  UIKit                               0x000000010ec9eaf8 -[UIControl touchesEnded:withEvent:] + 601
    12  UIKit                               0x000000010eb9f49b -[UIWindow _sendTouchesForEvent:] + 835
    13  UIKit                               0x000000010eba01d0 -[UIWindow sendEvent:] + 865
    14  UIKit                               0x000000010eb4eb66 -[UIApplication sendEvent:] + 263
    15  UIKit                               0x000000010eb28d97 _UIApplicationHandleEventQueue + 6844
    16  CoreFoundation                      0x00000001116a8a31 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
    17  CoreFoundation                      0x000000011169e95c __CFRunLoopDoSources0 + 556
    18  CoreFoundation                      0x000000011169de13 __CFRunLoopRun + 867
    19  CoreFoundation                      0x000000011169d828 CFRunLoopRunSpecific + 488
    20  GraphicsServices                    0x00000001132e9ad2 GSEventRunModal + 161
    21  UIKit                               0x000000010eb2e610 UIApplicationMain + 171
    22  Cyrano                              0x000000010c4ba3bf main + 111
    23  libdyld.dylib                       0x000000011236192d start + 1
    24  ???                                 0x0000000000000001 0x0 + 1
)
libc++abi.dylib: terminating with uncaught exception of type NSException
```

**Comment by abhinav4Jets** [comment]
Was there any solution to it , i am also stuck with this issue occurring ?

**Comment by Raikerian** [comment]
Hey @abhinav4Jets , have you tried our latest version that was recently released?

**Comment by abhinav4Jets** [comment]
Thanks for the Response @Raikerian  . I did and i guess it is 2.9.2. But as soon as i process [function -did send] method a crash occurs . I digged a little deep and it brought me here i.e Assert for Dialog id should be present in Memory of Dialog Arrays. Am i missing something like manually inserting dialog object when initiated from API hit or its done behind the scenes ? Any direction to solve it will be of great help.
